test/Test/Database/LSMTree/Normal/StateMachine.hs

```diff
@@ -591,8 +592,9 @@ instance Eq (Obs h a) where
       -- blob reference immediately after an update to the table, and if the SUT
       -- returns a blob, then that's okay. If both return a blob or both return
       -- an error, then those must match exactly.
-      (OEither (Right (OBlob (WrapBlob _))), OEither (Left (OId y)))
-        | Just Model.ErrBlobRefInvalidated <- cast y -> True
+      (OEither (Right (OVector vec)), OEither (Left (OId y)))
+        | Just (OBlob (WrapBlob _), _) <- V.uncons vec
+        , Just Model.ErrBlobRefInvalidated <- cast y -> True
```

Comment:
Why was this changed to the vector on the LHS? It means we're now only checking the first blob reference in the vector, not all of the blob references

Reply:
I think the situation this covers is when the model implementation of `RetrieveBlobs` returns an error, but the real successfully returns a vector. So it needs to deal with the vector.

> we're now only checking the first blob reference in the vector

We are just making sure it's a (non-empty) vector of blobs, which was probably noisy to do in a single pattern. But if the first element is a blob, I don't see how the others could be something else.

Reply:
You are quite right -- (my) old version was bad :P